Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs

```diff
@@ -1,32 +1,152 @@
+using System.Linq;
+using Content.Server.Administration.Logs;
 using Content.Server.Chemistry.Components;
-using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Chemistry;
+using Content.Shared.Chemistry.Dispenser;
+using Content.Shared.Containers.ItemSlots;
+using Content.Shared.Database;
 using Content.Shared.Emag.Systems;
 using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
 using Robust.Shared.Containers;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
 
 namespace Content.Server.Chemistry.EntitySystems
 {
+    /// <summary>
+    /// Contains all the server-side logic for reagent dispensers.
+    /// <seealso cref="ReagentDispenserComponent"/>
+    /// </summary>
     [UsedImplicitly]
-    public sealed class ReagentDispenserSystem : SharedReagentDispenserSystem
+    public sealed class ReagentDispenserSystem : EntitySystem
     {
+        [Dependency] private readonly AudioSystem _audioSystem = default!;
+        [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+        [Dependency] private readonly ItemSlotsSystem _itemSlotsSystem = default!;
+        [Dependency] private readonly UserInterfaceSystem _userInterfaceSystem = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+        [Dependency] private readonly EntityManager _entities = default!;
```

Comment:
all entitysystems have an implicit reference to `IEntityManager` through the `EntityManager` property, this is not needed